### PR TITLE
enhancement(datadog_agent source): Support v2 endpoint for series 

### PR DESCRIPTION
**Pull request by prognant**
Add v2 endpoint for series (protobuf encoded) support in the `datadog_agent` source. 

From an agent perspective this is currently an opt-in configuration flag (`use_v2_api.series: true` must be added to the agent config to turn it on) but it will soon be enabled by default, thus this change is mandatory to prevent an issue when the agent with the v2 series endpoint enabled by default will be released.

Improvement for this PR or a subsequent issue: emit a dogstatsd metrics to an agent around [here](https://github.com/vectordotdev/vector/blob/master/src/sources/datadog/agent/integration_tests.rs) an assert we got it right out of the `datadog_agent` source (agent config / startup is done [here](https://github.com/vectordotdev/vector/blob/master/scripts/integration/docker-compose.datadog-agent.yml) for this test).

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: src/sources/datadog/agent/tests.rs [hunk 1]
```diff
@@ -1462,3 +1454,199 @@ fn test_config_outputs() {
...
+            ddmetric_proto::metric_payload::MetricSeries {
+                resources: vec![ddmetric_proto::metric_payload::Resource {
+                    r#type: "host".to_string(),
+                    name: "random_host".to_string(),
+                }],
+                metric: "dd_gauge".to_string(),
+                tags: vec!["foo:bar".to_string()],
+                points: vec![
+                    ddmetric_proto::metric_payload::MetricPoint {
+                        value: 3.14,
```
[the remark this answers is not in the document]
Comment (prognant) [reply src/sources/datadog/agent/tests.rs:1479]: About that... I forgot all the advanced processing in the agent in terms of buffering & aggregation. And as the agent starts without a valid endpoint the agent forwarder defer some request for retry (keep in mind that the agent send a bunch of internal / system metrics every 10 sec), I tried to tweak the retry policy to get everything flush in a short time frame, while this approximately works another behaviour of the agent is problematic here, the aggregation layer can turn a single count(42) metrics into two metrics (count(42) and count(0)). I think that this integration test is not well suited for metrics. I'll create an issue pointing to this PR to follow up and investigate a better way to do it.

**Review by spencergilbert — APPROVED** [review]

**Comment by spencergilbert** [comment]
It looks like the `humio` soak might have failed/been broken? I know the soak needed to be updated when @pablosichert made encoding changes, but I don't _think_ that's the case here?

Maybe just needs master merged in again 🤷

File: src/sources/datadog/agent/metrics.rs [hunk 2]
```diff
@@ -182,7 +215,142 @@ fn decode_datadog_sketches(
...
+            (!serie.source_type_name.is_empty())
+                .then(|| tags.insert("source_type_name".into(), serie.source_type_name));
+            // As per https://github.com/DataDog/datadog-agent/blob/a62ac9fb13e1e5060b89e731b8355b2b20a07c5b/pkg/serializer/internal/metrics/iterable_series.go#L224
+            // serie.unit is omitted
+            match metric_payload::MetricType::from_i32(serie.r#type) {
+                Some(metric_payload::MetricType::Count) => serie
+                    .points
+                    .iter()
+                    .map(|dd_point| {
+                        Metric::new(
```
Comment (neuronull) [inline src/sources/datadog/agent/metrics.rs:285]: Would it be a good idea to include the parsing out of the namespace here in the v2 as was added recently for the v1?

https://github.com/vectordotdev/vector/blob/1df4d50ae20491c2a87e280aed7395e2ca79054b/src/sources/datadog/agent/metrics.rs#L413

(per this PR) : https://github.com/vectordotdev/vector/pull/13176/files

Comment (jszwedko) [reply src/sources/datadog/agent/metrics.rs:285]: Agreed, this got automerged, but we should add this namespace parsing here too.

Comment (neuronull) [reply src/sources/datadog/agent/metrics.rs:285]: Hmmm, new issue then ? I can open it.

Comment (prognant) [reply src/sources/datadog/agent/metrics.rs:285]: I did it in #13376

**Pull request merged**